src/document.rs

```diff
@@ -39,18 +39,71 @@ fn read_file<P: AsRef<Path>>(path: P) -> Result<String> {
     Ok(text)
 }
 
+fn yaml_btreemap_to_hashmap(input: &BTreeMap<Yaml, Yaml>) -> HashMap<String, Value> {
+    let mut result = HashMap::new();
+    for (yaml_key, yaml_value) in input {
+        if let Yaml::String(ref s) = *yaml_key {
+            let liquid_value = yaml_to_liquid(yaml_value).expect("Value must be liquid data");
+            result.insert(s.to_owned(), liquid_value);
+        } else {
+            panic!("Key in yaml dictionary must be string");
+        }
+    }
+    result
```

Comment:
I can understand this feeling weird.  The Rust community's coding conventions are very terse and while I don't agree with it in some cases like this, I'm slowly getting used to it.